maven/lib/dependabot/maven/utils/auth_headers_finder.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Maven
+    module Utils
+      class AuthHeadersFinder
```

Comment:
This is currently the same between the Maven and Gradle implementation right? Wondering if it makes sense to hoist this up to the `common` package and reuse. Should we ever expect these implementations to deviate?

Reply:
That was my thought as well, but I was not sure if this was common enough, as it is very specific for maven repos. There is a natural dependency from Gradle to Maven, as Gradle uses Maven dependencies. Would it make sense to let the gradle module depend on the maven module?
Otherwise it should probably be moved to the common package. Implementation should not deviate, as long as it concerns maven repositories.

Reply:
I have taken the liberty of adding maven as a dependency on the gradle module. I will move it to the common module if this is preferred. In that case suggestions for where it should be placed are welcome.

Reply:
I think adding maven as a dependency is fine 👍 . I wonder if there are more things we can reuse between the two, but let's leave that as an exercise for the future